index.js

```diff
@@ -242,7 +242,10 @@ function normalizeArguments(url, opts) {
 
 		opts.method = opts.method || 'POST';
 
-		if (isPlainObj(body)) {
+		if (isStream(body) && typeof body.getBoundary === 'function') {
+			// Special case for https://github.com/form-data/form-data
+			opts.headers['content-type'] = opts.headers['content-type'] || 'multipart/form-data; boundary=' + body.getBoundary();
```

Comment:
`content-type` is not the only header, that can be returned from `getHeaders` function. This line should contain call to `getHeaders` and assign user supplied headers over it.